taskvine/src/manager/vine_manager.c

```diff
@@ -3746,6 +3743,18 @@ static vine_task_state_t change_task_state( struct vine_manager *q, struct vine_
 			vine_task_set_result(t, VINE_RESULT_UNKNOWN);
```

Comment:
It seems that all of these cases take on the responsibility of **pushing** to a data structure, but not removing from the old one.  With the exception of `RETRIEVED` which does both.  Is there a reason for the inconsistency?

Reply:
It could be worth some cleanup work. But the motivation for how it is now is that we remove a task from the ready list when we commit it to the worker by calling list_pop_tail, which can be completed without having to seek through the task list.

In theory we could call list_pop_tail inside change_task_state since nothing unexpected should happen between the two events, but it would not be as clear in the code to know the task we just scheduled was positioned at the end of the list.

We are also ignoring the VINE_TASK_RUNNING state transition inside change_task_state, since there has been no apparent utility for a separate running_list data structure. So when a task changes to VINE_TASK_WAITING_RETRIEVAL, it is not currently in any of the lists so it does not need to be removed from anywhere.